docs/architecture/blazor-for-web-forms-developers/hosting-models.md

```diff
@@ -1,9 +1,80 @@
+
+## How to choose the right Blazor hosting model
+
+As described in the [Blazor hosting model docs](https://docs.microsoft.com/aspnet/core/blazor/hosting-models#server-side), the different Blazor hosting models have different tradeoffs.
+
+The Blazor WebAssembly hosting model has these benefits:
+
+* There's no .NET server-side dependency. The app is fully functioning after downloaded to the client.
+* Client resources and capabilities are fully leveraged.
+* Work is offloaded from the server to the client.
+* An ASP.NET Core web server isn't required to host the app. Serverless deployment scenarios are possible (for example, serving the app from a CDN).
+
+The downsides of the Blazor WebAssembly hosting model are:
+
+* The app is restricted to the capabilities of the browser.
+* Capable client hardware and software (for example, WebAssembly support) is required.
+* Download size is larger, and apps take longer to load.
+* .NET runtime and tooling support is less mature. For example, limitations exist in [.NET Standard](/dotnet/standard/net-standard) support and debugging.
+
+Conversely. the Blazor Server hosting model offers these benefits:
+
+* Download size is significantly smaller than a client-side app, and the app loads much faster.
+* The app takes full advantage of server capabilities, including use of any .NET Core compatible APIs.
+* .NET Core on the server is used to run the app, so existing .NET tooling, such as debugging, works as expected.
+* Thin clients are supported. For example, server-side apps work with browsers that don't support WebAssembly and on resource-constrained devices.
+* The app's .NET/C# code base, including the app's component code, isn't served to clients.
+
+The downsides to the Blazor Server hosting model are:
+
+* Higher UI latency. Every user interaction involves a network hop.
+* There's no offline support. If the client connection fails, the app stops working.
+* Scalability is challenging for apps with many users. The server must manage multiple client connections and handle client state.
+* An ASP.NET Core server is required to serve the app. Serverless deployment scenarios aren't possible (for example, serving the app from a CDN).
+
+While these tradeoffs may seem daunting, the good news is that you can always change your mind later. Regardless of which Blazor hosting model you choose the component model is *the same*. This means that the same components can in principle be used with either hosting model. Your app code can stay the same. While it is possible to author components that can only run on the server or in the browser, it's generally good practice to put appropiate abstractions in place so that your components stay hosting model agnostic. That way, you can easily switch your app to use a different hosting model if needed.
```

Comment:
Merriam suggests there's a hyphen in https://www.merriam-webster.com/dictionary/trade-off 

```suggestion
While these trade-offs may seem daunting, the good news is that you can always change your mind later. Regardless of which Blazor hosting model you choose, the component model is *the same*. This means that the same components can in principle be used with either hosting model. Your app code can stay the same. While it is possible to author components that can only run on the server or in the browser, it's generally good practice to put appropriate abstractions in place so that your components stay hosting model agnostic. That way, you can easily switch your app to use a different hosting model if needed.
```